test/e2e/testdata/custom-container-security-contex-userid.yaml

```diff
@@ -0,0 +1,44 @@
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: EnvoyProxy
+metadata:
+  name: custom-container-security-contex-userid-envoyproxy
+  namespace: gateway-conformance-infra
+spec:
+  provider:
+    type: Kubernetes
+    kubernetes:
+      envoyDeployment:
+        container:
+          securityContext:
+            runAsUser: 65534
```

Comment:
the related securityContext is tied to EG Deployment, not Envoy

Reply:
This is envoy deployment:

```yaml
apiVersion: [gateway.envoyproxy.io/v1alpha1](http://gateway.envoyproxy.io/v1alpha1)
kind: EnvoyProxy
metadata:
  name: custom-container-security-contex-userid-envoyproxy
  namespace: gateway-conformance-infra
spec:
  provider:
    type: Kubernetes
    kubernetes:
      envoyDeployment:     # envoy deployment, for both the envoy and shutdown manager Security Context
        container:
          securityContext:
            runAsUser: 65534
            runAsGroup: 65534
```